### PR TITLE
Set ssl_context.check_hostname to False in create_ssl_context().

### PR DESCRIPTION
**Pull request by kitprice**
This should resolve the current issue [#57582](https://github.com/home-assistant/core/issues/57582) in Home Assistant.

### CODE REVIEW

**Comment by meyerrj** [comment]
> http://theautomatic.net/2019/01/19/scraping-data-from-javascript-webpage-python/ <-- this is the best i could find (admittedly, i don't even know how best to googlize this kind of question)

Looks like this uses a module called requests_html:  https://requests-html.kennethreitz.org/ 

This module downloads chromium to do the JavaScript processing/rendering. This could be an issue with implementations in other services such as Home Assistant, as it may result in platform-specific issues related to chromium... Given that, I'm not sure this is a route that would be worth investing time in, unfortunately.

**Comment by johnavich** [comment]
So, i just got off the call with SMT, and they stated that their security team looked at the request i was making, and the system was working as designed.  They do not support scripted access to their system accept through the System API service.smartmetertexas.net.  

Our best bet in the mean time would be to use some method like BeautifulSoup or requests_html to render the javascript and get the appropriate cookies.

On that note, I plan on filing a dispute with the PUCT regarding the requirements, as they were intended for large scale corporations like Competitive Service Providers, and Retail Electric Providers to get mass updates for their various customers.  PUCT Order 47472 states they need a Static IP to whitelist, and a publicly signed SSL Certificate to be authenticated against.  This does not include most residential ISP customers, thus most of us.  If anyone has any interest, or knowledge of such things, please reach out to me.

**Comment by adamfeldman** [comment]
I don't want to distract too much from this PR thread, am happy to move the below discussion into an issue or elsewhere upon request.

@johnavich I'm interested in helping draft the PUCT dispute and filing a copy myself.

With regard to the static IP requirement for using the [official API (pages 15-16)](https://www.smartmetertexas.com/Smart_Meter_Texas_Interface_Guide.pdf), I've thought about setting up a secure proxy of sorts for us all to share. The SMT Interface Guide anticipates such a scenario with "Independent IT Service Providers".

The proxy would be locked-down to only send traffic to SMT. Likely it could be hosted cheaply on a DO droplet or similar. People could gain access to it upon request, and authenticate themselves with an API token or such.

**Comment by johnavich** [comment]
@adamfeldman I was thinking exactly this same thing.  setting up an API endpoint for us all, using my linode box.  Glad to know i wasn't the only one thinking it

**Comment by grahamwetzler** [comment]
Unfortunately that seems to be the only real option going forward. I wonder if there's enough interest if this could be coordinated through Nabu Casa and Home Assistant cloud?

**Comment by adamfeldman** [comment]
@grahamwetzler I very much like the idea of Nabu Casa handling it. 

I would be happily surprised if they accepted managing the complexity involved...if I were asking them, I'd ask in context of "SMT must not be the only HA integration that has some ops complexity – static IP etc – around talking to a service provider's API"

**Comment by johnavich** [comment]
I feel like this could become an actual conversation, much too hard to follow in a single thread. Does anyone have a discord server they'd like offer up so we can chat more real time?

**Comment by johnavich** [comment]
https://discord.gg/yv8ubgPk <-- n/m, we can use mine.

**Comment by johnavich** [comment]
i know this case is "old" and most of us have implemented our own method for solving this issue. Has anyone looked at https://github.com/klaasnicolaas/home-assistant-glow yet?  it could probably replace SMT entirely, and i would prefer it. but i need to know how one would utilize it in an outdoor situation :(

**Comment by mrand** [comment]
> i know this case is "old" and most of us have implemented our own method for solving this issue. Has anyone looked at https://github.com/klaasnicolaas/home-assistant-glow yet? it could probably replace SMT entirely, and i would prefer it. but i need to know how one would utilize it in an outdoor situation :(

It isn't obvious to me that the Oncor meters have a pulse output.  There is a very clear LCD screen that has a "chasing" LCD indicator which might or might not be something similar, but would presumably require OCR to detect it.  If you go to that trouble, might as well just OCR the kWh straight off the meter.  Annoying part is that it isn't a static display.

An easier solution would be to buy a HAN (Zigbee bridge) and read the meter directly, but apparently Oncor has found a way to [stop supporting even that](https://community.home-assistant.io/t/rainforest-eagle-200-energy-gateway-does-you-have-one/21253/81?u=mrand)! :-(

**Comment by johnavich** [comment]
@mrand your entirely correct, I missed the method of reading and what the device is doing. This wouldn't work for us without also having an emitting diode to tell the meter to respond :-(

Oh well... it was worth a try
